logic/src/commonTest/kotlin/com/wire/kalium/logic/data/register/RegisterAccountRepositoryTest.kt

```diff
@@ -118,35 +118,60 @@ class RegisterAccountRepositoryTest {
         val ssoId = with(TEST_USER.ssoID) {
             this?.let { SsoId(scimExternalId = it.scimExternalId, subject = it.subject, tenant = it.tenant) }
         }
+        val cookieLabel = "COOKIE_LABEL"
         val authTokens = with(SESSION) {
             AuthTokens(
                 userId = UserId(userId.value, userId.domain),
                 accessToken = accessToken,
                 refreshToken = refreshToken,
-                tokenType = tokenType
+                tokenType = tokenType,
+                cookieLabel = cookieLabel
             )
         }
         val expected = Pair(ssoId, authTokens)
 
         given(registerApi).coroutine {
             register(
-                RegisterApi.RegisterParam.PersonalAccount(email, code, name, password)
+                RegisterApi.RegisterParam.PersonalAccount(
+                    email = email,
+                    emailCode = code,
+                    name = name,
+                    password = password,
+                    cookieLabel = cookieLabel
+                )
             )
         }.then { NetworkResponse.Success(Pair(TEST_USER, SESSION), mapOf(), 200) }
         given(idMapper).invocation { toSsoId(TEST_USER.ssoID) }.then { ssoId }
         given(sessionMapper).invocation { fromSessionDTO(SESSION) }.then { authTokens }
 
-        val actual = registerAccountRepository.registerPersonalAccountWithEmail(email, code, name, password)
+        val actual = registerAccountRepository.registerPersonalAccountWithEmail(
+            email = email,
+            code = code,
+            name = name,
+            password = password,
+            cookieLabel = cookieLabel
+        )
 
         assertIs<Either.Right<Pair<SsoId?, AuthTokens>>>(actual)
         assertEquals(expected, actual.value)
 
-        verify(registerApi).coroutine { register(RegisterApi.RegisterParam.PersonalAccount(email, code, name, password)) }
+        verify(registerApi).coroutine {
+            register(
+                RegisterApi.RegisterParam.PersonalAccount(
+                    email = email,
+                    emailCode = code,
+                    name = name,
+                    password = password,
+                    cookieLabel = cookieLabel
```

Comment:
Not sure if I fully understand these tests. Does this mean that the specified `cookieLabel` that we pass upon client registration should be also returned untouched by the backend?

Reply:
the label is passed from the clients to BE, and we need to make sure that we store the same label that we used to login/register. and the same label is also used for client registration.

this way when a client get deleted it's cookies will also get deleted